modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointDirtyPages.java

```diff
@@ -229,15 +242,15 @@ public PersistentPageMemory pageMemory() {
          * Returns the size of the view.
          */
         public int size() {
-            return toPosition - fromPosition + 1;
+            return toPosition - fromPosition;
         }
 
         private CheckpointDirtyPages owner() {
             return CheckpointDirtyPages.this;
         }
 
-        private boolean isToPositionLast() {
-            return toPosition == dirtyPages.get(regionIndex).dirtyPages.pageIds.length - 1;
+        private boolean needsNextRegion() {
+            return toPosition == dirtyPages.get(regionIndex).dirtyPages.pageIds.length;
```

Comment:
Why don't we store `dirtyPages.get(regionIndex)` precalculated? No one needs this extra lookup on every access

Reply:
A reference weighs more than a (**int**) primitive, and it's easier for the garbage collector with a primitive. Do you insist on removal?

Reply:
What does it have to do with GC? Your object will be 4 bytes bigger. No new objects required. You can store both index and dirty pages collection, it's not a problem

Reply:
But why? In general objects are not a problem worth going down the path of pure OOP.

Reply:
nextPartitionView - here I need this index, maybe I shouldn't remove it?

Reply:
You can have both index and collection, it's fine